### ISSUE TITLE
Historical backtracking

### ISSUE THREAD

**Issue by Liu-ErMeng** [flameshot-org/flameshot#3350]
### Feature Description

I think it will be great if flameshot have a feature to backtrack the historical screenshots like [snipaste](https://www.snipaste.com/) under windows.

**Comment by mmahmoudian** [comment]
Thanks for making the feature request, but none of the devs use Windows, so would would be nice if you can explain in details what exactly you are looking for to have. I mean just elaborate in more details. Thanks.

**Comment by Liu-ErMeng** [comment]
> Thanks for making the feature request, but none of the devs use Windows, so would would be nice if you can explain in details what exactly you are looking for to have. I mean just elaborate in more details. Thanks.

snipaste got key bindings`, ` to "Switch to the previous snip record" and ` .` to "Switch to the next snip record", which let you backtrack the snip records history, and you can remodify the older snip record, copy, save, etc. 
It's a wonderfull feature, so I think it will be great if flameshot support this too.

https://docs.snipaste.com/key-bindings

**Comment by mmahmoudian** [comment]
But what does the "snip record history" even mean? Is it the last positions of the selected region? Is it the screenshots you have taken?

Can you record a short video or somehow illustrate what you are suggesting?

**Comment by Liu-ErMeng** [comment]
> But what does the "snip record history" even mean? Is it the last positions of the selected region? Is it the screenshots you have taken?
> 
> Can you record a short video or somehow illustrate what you are suggesting?
yes, there is [video](https://youtu.be/ukN9VqFYCSE?t=325) demostrate how the  "replay snipaste history" work.

**Comment by mmahmoudian** [comment]
I believe this is best to be dealt with via a plugin. This is because:
1. in Flameshot we try our best to avoid recording anything from user for privacy reasons
2. if we just store the path, and the user move/delete the file it will break the flow
3. if we store a duplicate of every screenshot, then
    3.1. more disk space will be used (twice as much)
    3.2. we should also provide various mechanisms for the user to manage this "storage" as well (e.g wiping all, deleting specific ones, changing orders, ...)
4. this is a bit far from a "screenshot" paradigm and get into "gallery management".

Of course, as always, this is just my personal opinion and we can wait and see if other devs have counter arguments.

For now I will label it with [plugin](https://github.com/flameshot-org/flameshot/issues?q=is%3Aopen+is%3Aissue+label%3APlugin) label.
